packages/ui-app/src/InputNumber.tsx

```diff
@@ -0,0 +1,191 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { BareProps, I18nProps } from './types';
+
+import BN from 'bn.js';
+import React from 'react';
+import isString from '@polkadot/util/is/string';
+
+import classes from './util/classes';
+import Input, { KEYS, KEYS_PRE, isCopy, isCut, isPaste, isSelectAll } from './Input';
+import translate from './translate';
+
+type Props = BareProps & I18nProps & {
+  bitLength?: number,
+  defaultValue?: string,
+  isError?: boolean,
+  label?: any,
+  maxLength?: number,
+  onChange: (value: BN) => void,
+  placeholder?: string,
+  withLabel?: boolean
+};
+
+type State = {
+  isPreKeyDown: boolean,
+  isValid: boolean,
+  previousValue: string
+};
+
+const BIT_LENGTH_32 = 32; // normal numbers
```

Comment:
Any reason we can't just use 32, 128 where we pass it in?

Elsewhere in the API (encoding/decoding), have actually defined this type of thing in the following manner:

```
type BitLength = 32 | 64 | 128 | 256;

// default here
function foo (bitLength?: BitLength = 128) {
 ...
}

foo(32); // works
foo(64); // works
foo(121); // does not work, TS checks shows an error
```

Reply:
@jacogr I've pushed a commit with these changes.

I also created a TS `enum` in enum.ts (since enums don't go in .d.ts files) and used it as shown below, but it wouldn't recognise it in the unit tests

```
// enum.ts
export declare const enum BitLengthOption {
  CHAIN_SPEC = 128,
  NORMAL_NUMBERS = 32
}

// InputNumber.tsx
const DEFAULT_BITLENGTH: BitLength = BitLengthOption.NORMAL_NUMBERS;

// Transfer.tsx / Balance.tsx
const DEFAULT_BITLENGTH: BitLength = BitLengthOption.CHAIN_SPEC;
```

So I reverted to just using key/value pairs, which worked and passes the unit tests. But I wasn't sure where to put it (i.e. if I put it in Balance.tsx, then it would be strange to import it and use BitLengthOption in Transfer.tsx where we are importing InputNumber directly...), so I've just left it in enum.ts for the moment.

```
// enum.ts
export const BitLengthOption = {
  CHAIN_SPEC: 128 as BitLength,
  NORMAL_NUMBERS: 32 as BitLength
};
```

I like doing this because I find it more readable (and avoids me leaving a comments like `// 32 is normal number, 128 is chain spec` above the type definition of BitLength for future reference, but I'm also happy to remove it